src/models/user.js

```diff
@@ -1,15 +1,55 @@
 import crypto from 'crypto'
 import { Model } from 'sequelize'
+import jwtBuilder from '../JWTBuilder'
+import UsernameNotFoundError from '../errors/UsernameNotFoundError'
+import InvalidPasswordError from '../errors/InvalidPasswordError'
 
-export default (sequelize, DataTypes) => {
+export function buildUserModel (sequelizeModels) {
   class User extends Model {
-    async passwordMatches (value, models = sequelize.models) {
+    static async signin (username, password) {
+      const user = await User.findByUsername(username)
+
+      await user.validatePassword(password)
+
+      const jwt = await user.generateJWT()
+
+      return { user, jwt }
+    }
+
+    static async findByUsername (username) {
+      const user = await User.findOne({ where: { username } })
+
+      if (!user) {
+        throw new UsernameNotFoundError('Could not find a User with username: ' + username)
```

Comment:
This is not very secure. You're given the "hacker" a clue about this user, so now he only has to guess its password. I don't remember where, but I've read that it's better to say "invalid username OR password" so you don't give any clue about the user.